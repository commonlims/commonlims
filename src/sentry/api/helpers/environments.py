from __future__ import absolute_import

from sentry.api.exceptions import ResourceDoesNotExist
from sentry.models import Environment


def get_environments(request, organization):
    requested_environments = set(request.GET.getlist('environment'))

    if not requested_environments:
        return []

    environments = list(Environment.objects.filter(
        organization_id=organization.id,
        name__in=requested_environments,
    ))

    if set(requested_environments) != set([e.name for e in environments]):
        raise ResourceDoesNotExist

    return environments
