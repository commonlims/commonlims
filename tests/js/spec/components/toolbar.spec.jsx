import React from 'react';
import Toolbar from 'app/components/toolbar';

import {render} from 'app-test/helpers/render';

import serializer from 'jest-emotion';
expect.addSnapshotSerializer(serializer);

describe('Toolbar', function () {
  let sandbox;

  beforeEach(function () {
    sandbox = sinon.sandbox.create();
  });

  afterEach(function () {
    sandbox.restore();
  });

  it('renders', function () {
    const rendered = render(
      <Toolbar>
        <div />
      </Toolbar>
    );
    expect(rendered).toMatchSnapshot();
  });
});
