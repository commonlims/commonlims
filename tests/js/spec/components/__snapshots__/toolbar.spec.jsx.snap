// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders 1`] = `
.emotion-0 {
  background: #fbfbfc;
  border: 1px solid #E2DBE8;
  border-bottom: none;
  border-radius: 3px 3px 0 0;
  box-shadow: 0 1px 0 #E2DBE8;
  margin: 0;
}

<div
  className="emotion-0 emotion-1"
>
  <div />
</div>
`;
